WireDomain/Sources/WireDomain/Synchronization/PushChannelState.swift

```diff
@@ -37,15 +46,15 @@ struct PushChannelState: PushChannelStateProtocol {
         self.fileContext = SafeFileContext(fileURL: url)
     }
 
-    func isOpen() -> Bool {
-        fileContext.isLocked()
-    }
-
-    func markAsOpen() {
-        fileContext.acquireDirectoryLock()
+    func markAsOpen() throws {
+        if !fileContext.tryAcquireLock() || Self.locked {
```

Comment:
in order to avoid race condition we try to acquire the lock instead of checking if isLocked before. Also acquiring the lock within the same process will always succeed hence the check with the locked static property